File: nodejs/lib/foursquare-proxy.js
// StateHQ
var LAT = 51.508857899999995;
var LONG = -0.195678;

var foursquare = require("foursquare-js")({
  secrets: {
    clientId: "AV03ZKZYEYRCHDPEIJN4QDMCZV30M5UM14XWSONSNS3YQFNV",
    clientSecret: "HICXJ5YBXQRJGH0GW0DIYXGSNGAKYXWCK5WL00EWMPHRE1CJ",
    redirectUrl: "n/a"
  },

  foursquare: {
    version: "20130606"
  }
});

exports.searchVenues = function(query, callback){
  foursquare.Venues.search(LAT, LONG, null, { query: query }, null, callback);
};

File: lib/venue_finder/foursquare_proxy.rb
require 'foursquare2'

module VenueFinder
  class FoursquareProxy
    CLIENT_ID = 'AV03ZKZYEYRCHDPEIJN4QDMCZV30M5UM14XWSONSNS3YQFNV'
    CLIENT_SECRET = 'HICXJ5YBXQRJGH0GW0DIYXGSNGAKYXWCK5WL00EWMPHRE1CJ'
    API_VERSION = '20130101'

    LAT = 51.508857899999995
    LONG = -0.195678

    def initialize
      @client = Foursquare2::Client.new(:client_id => CLIENT_ID, :client_secret => CLIENT_SECRET, :version => API_VERSION)
    end

    def search_venues query
      @client.search_venues(:ll => "#{LAT},#{LONG}", :query => query).groups.first.items
    end
  end
end
